XTree/XTree/Modules/Main/MainState.swift

```diff
@@ -0,0 +1,22 @@
+import Combine
+
+final class MainState: ObservableObject {
+    @Published var filters = FiltersViewState()
+    @Published var toolbar = ToolBarState()
+    @Published var isInDropArea = false
+    @Published var tree: TreeViewState?
+
+    private var ignoresChanges = false
+    private var bag = Set<AnyCancellable>()
+
+    func onToolbarChange(_ onChange: @escaping (@escaping () -> Void) -> Void) {
+        guard bag.isEmpty else { return }
+        $toolbar.sink(receiveValue: { [weak self] _ in
+            guard let self = self, !ignoresChanges else { return }
+            self.ignoresChanges = true
+            onChange {
+                self.ignoresChanges = false
+            }
+        }).store(in: &bag)
+    }
+}
```

Comment:
It looks complicated to me 😅.
Why do we need this? Everything worked fine without these changes on my MacBook.

And by the way, I don't think it's related to the PR topic.

Reply:
The `.onChange()` api is not available on Mac OS 13.0 thus i was trying to make it work. I agree that it is totally complicated than should be. But i didn't come up with a better solution :( So any ideas on that one?
<img width="517" alt="Screenshot 2024-05-14 at 23 23 01" src="https://github.com/swiftyfinch/xtree/assets/36343782/1f8c4241-a3aa-4283-b438-53e473b5f9fd">

Reply:
I will fix this in the main brach